### PR TITLE
react-components: robot panel

### PR DESCRIPTION
**Pull request by vallq**
## What's new
Robot Panel component based off from Task Panel
robot panel ui
- will be removing the refresh button

## Self-checks

- [x] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [x] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

## Discussion
1. I'm still not sure how to use the `api_server`'s implementation of task progress in `react-components` because what react-components use is technically just dummy data. should this be taken as a prop and only implemented in the dashboard?

~2. The original idea was to also have 'Est End Time' be a clock value in 12hour format, and likewise, I'm not sure how this can be done. using `rosTimeToJs` in `lib/utils.ts`?~

### CODE REVIEW

File: packages/react-components/lib/robots/robot-panel.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+              rowsPerPageOptions: [10],
+              page,
+              onChangePage: (_ev, newPage) => setPage(newPage),
+            }}
+            onRobotClick={(_ev, robot) => setSelectedRobot(robot)}
+            onRefreshClick={onRefreshClick}
+          />
+        </Grid>
+        <Paper className={classes.detailPanelContainer}>
+          {selectedRobot ? <RobotInfo robot={selectedRobot} /> : <NoSelectedRobot />}
```
Comment (matiasbavera) [inline packages/react-components/lib/robots/robot-panel.tsx:68]: Does it improve the performance if you store this on a `React.useMemo` or using `React.memo` on `RobotInfo ` component?

**Review by matiasbavera — CHANGES_REQUESTED** [review]
Hi Val, left a couple of comments, HTH

**Review by koonpeng — COMMENTED** [review]
I'm going to push some changes to this branch that adds some features to make query the tasks/robots easier.

When using the new task query api, you can pass the list of tasks as an object, you can get the task_id with RobotState.task_id. On the dashboard side you can

1. Use `/fleets` to get a list of fleets.
2. Use `/fleets/{name}/state` to get the state of each fleet (and robots), this is inefficient as it means sending a request for each fleet, but it is the best we have now.
3. Use `/tasks?task_id=a,b,c` to get the task summary of each task.

**Comment by koonpeng** [comment]
from the other pr:
> oh okay!
other than the current tasks, is it possible to retrieve all assigned and active tasks? (completed/failed tasks are not needed). because there is an 'assigned tasks' section in the robot-info.tsx

oh I see, the information in robot state is not enough, I think disregard the above guide. Just use the task query api the find the list of assigned tasks for a robot. The new additions allow you to query by multiple robot/fleet/states so you should be able to get the required data.

File: packages/react-components/lib/robots/utils.ts [hunk 2]
```diff
@@ -29,3 +29,53 @@ export function robotModeToString(robotMode: RmfModels.RobotMode): string {
...
+  robots: RmfModels.RobotState[],
+  tasks: RmfModels.TaskSummary[],
+): VerboseRobot[] {
+  const removableTaskStates = [
+    RmfModels.TaskSummary.STATE_ACTIVE,
+    RmfModels.TaskSummary.STATE_PENDING,
+    RmfModels.TaskSummary.STATE_QUEUED,
+  ];
+  const robotsWithAssignedTasks = robots.map((robot) => {
+    const assignedTasks = tasks.filter((task) => {
```
Comment (koonpeng) [inline packages/react-components/lib/robots/utils.ts:73]: I think we should add a TODO to move this to the server. The server will have a robots endpoints returning the equivalent of `VerboseRobot`.

File: packages/react-components/lib/robots/robot-table.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  }
+}
+
+export interface RobotTableProps extends PaperProps {
+  /**
+   * The current list of robots to display, when pagination is enabled, this should only
+   * contain the robots for the current page.
+   */
+  tasks: RmfModels.TaskSummary[];
+  robots: RmfModels.RobotState[];
```
Comment (koonpeng) [inline packages/react-components/lib/robots/robot-table.tsx:128]: Could this take in a `VerboseRobot` instead?

**Comment by koonpeng** [comment]
I realised the robots endpoints will need more work to get the required info, so I think I will revert the api-server changes and work on it on another branch.

**Comment by vallq** [comment]
updated branch: `dashboard/robot-page`

**Pull request closed without merging**